test/controllers/auth/SelfAssessmentStatusActionSpec.scala

```diff
@@ -107,47 +113,88 @@ class SelfAssessmentStatusActionSpec
 
   "A user without an SA enrolment must" - {
     "when CitizenDetails has a matching SA account" - {
-      "and they have an enrolment on another credential" - {
-        "return WrongCredentialsSelfAssessmentUser" in {
-          implicit val request = createAuthenticatedRequest(None)
-
-          val saUtr = SaUtr("1111111111")
+      "and the cache does not contain an SA user" - {
+        "and they have an enrolment on another credential" - {
+          "return WrongCredentialsSelfAssessmentUser" in {
+            implicit val request = createAuthenticatedRequest(None)
+
+            val saUtr = SaUtr("1111111111")
+
+            when(enrolmentsCachingService.getSaUserTypeFromCache()(any(), any()))
+              .thenReturn(Future.successful(None))
+            when(enrolmentsCachingService.addSaUserTypeToCache(any[SelfAssessmentUserType]())(any(), any()))
+              .thenReturn(Future.successful(CacheMap("id", Map.empty)))
```

Comment:
A lot of mocking going on, when this happens it suggests we're just testing implementation rather than behaviour. 

I'll re-review this bit when the enrolmentsCachingService logic is updated (as mentioned above)